src/main/java/com/github/kagkarlsson/scheduler/ScheduledExecution.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) Gustav Karlsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.kagkarlsson.scheduler;
+
+
+import com.github.kagkarlsson.scheduler.task.Execution;
+import com.github.kagkarlsson.scheduler.task.TaskInstance;
+
+public class ScheduledExecution {
+
+    private final Execution execution;
+
+    public ScheduledExecution(Execution execution) {
+        this.execution = execution;
+    }
+
+    public TaskInstance getTaskInstance() {
+        return execution.taskInstance;
+    }
+

```

Comment:
Didn't you say your handle was in the data of the execution? Wouldn't you want to expose that as well for your requirement?  (not sure how I would do that though, since task is generic. you would have to do a check on task-name, and have a method like `<T> T getData(Class<T> dataClass)` to get the deserialized data)

For your requirement, maybe it would make more sense to get all the executions for a specific task only, then the data-class is the same for all? 

Up for discussion..

Reply:
Good point, because I was actually planning to filter on task name and type cast it on the client side. So I do agree that it seems desirable to get all the executions for a specific task. So maybe something like the following?
```java
public class ScheduledExecution<DATA_TYPE> {

    private final Execution execution;

    public ScheduledExecution(Execution execution) {
        this.execution = execution;
    }

    public TaskInstance getTaskInstance() {
        return execution.taskInstance;
    }

    public DATA_TYPE getData() {
        return (DATA_TYPE) this.execution.taskInstance.getData();
    }
}
```
Where the `Scheduler` could have both:
```java
public <T> List<ScheduledExecution<T>> getScheduled() {}
public <T> List<ScheduledExecution<T>> getScheduled(String taskName) {}
```

The client could then do:
```java
// Get all without type identification 
// Here the client could still type-cast if needed
// i.e. String data = (String) scheduler.getScheduled().get(0).getData();
List<ScheduledExecution<Object>> executions1 = scheduler.getScheduled();

// Seems less unlikely, but it could be that all executions are of 1 specific data type.
List<ScheduledExecution<String>> executions2 = scheduler.getScheduled();

// Get by specific task name.
List<ScheduledExecution<String>> executions3 = scheduler.getScheduled(TASK);
```

Let me know what you think and I will get it implemented :-)

Reply:
With the consumer-suggestion this becomes something like this I suppose?

```
List<ScheduledExecution<SomeId>> all = new ArrayList<>();
scheduler.<SomeId>getScheduled(TASK_NAME, add:add)

all.stream()
        .filter(data -> data.getData().getSomeId().equals(some_other_id))
        .map(ScheduledExecution::getTaskInstance)
        .forEach(scheduler::cancel);```

Reply:
And also not sure if maybe we should send in the dataclass as a parameter. Would be easier to document..

`void getScheduled(Consumer<ScheduledExecution<Object>> executionConsumer)`
`<T> void getScheduledForTask(String taskName, T dataClass, Consumer<ScheduledExecution<T>> executionConsumer)` 

This would be a bit prettier if I had some sort of static `TaskDescriptor` class, holding name and dataclass-type. A refactoring I am considering, but not prio yet..  (out of scope here)

Reply:
Yeah you are right, I didn't use the consumer in this example.

You could have two `getScheduled()`-methods one with the consumer and thus no return type and one without the consumer with a `List<>` as return type.. However, one could also implement the consumer like (without the need of adding it to a list all together):
```java
scheduler.<SomeId>getScheduled(ActivateBeaconTask.CREATE_BEACON_TASK, scheduledExecution -> {
    if (scheduledExecution.getData().getSomeId().equals(some_other_id)) {
        scheduler.cancel(scheduledExecution.getTaskInstance());
    }
});
```
so I leave that one up to you ;-). I will also get the same exposed on the `SchedulerClient`-interface.

Reply:
True, lets go with only the consumer-version for now, and consider adding the list-version if there is popular demand :)

Reply:
Updated the PR with some commits, I think I've implemented as discussed. Let me know if you have any more feedback.

Reply:
What about my comment about sending in a `T dataClass`? I think it's a bit clearer and explicit that way. (until I implement a sort of static task-descriptor)

```
void getScheduledExecutions(Consumer<ScheduledExecution<Object>> executionConsumer);

<T> void getScheduledExecutionsForTask(String taskName, T dataClass, Consumer<ScheduledExecution<T>> executionConsumer)```

Reply:
Just some small adjustment, but I think we can merge this soon :)

Reply:
`Class<T> dataClass` then surely instead of `T dataClass`?

I could then also have the data class passed to the `ScheduledExceution` where I can do some sort of "type checking", i.e.
```java
public DATA_TYPE getData() {
    if (dataClass.isInstance(this.execution.taskInstance.getData())) {
        return (DATA_TYPE) this.execution.taskInstance.getData();
    }
    
    throw new ???Exception();
}
```
Could be either a checked or unchecked exception.. do you think this would be useful?

Reply:
* Yes, you are correct! `Class<T>`. A `RuntimeException` would be nice, `DataClassMismatchException` or something like that

* Could you also change the `getTaskInstance` method to return a `TaskInstanceId` instead, so that we only have a single way of accessing the data?

* And lastly, it seems reasonable to also expose the execution-time on the `ScheduledExecution`